src/LVMIndex.cpp

```diff
@@ -238,16 +238,19 @@ class GenericIndex : public LVMIndex {
         Entry a = order.encode(low.asTuple<Arity>());
         Entry b = order.encode(high.asTuple<Arity>());
         // Transfer upper_bound to a equivalent lower bound
-        for (size_t i = Arity - 1; i >= 0; --i) {
+        bool fullIndexSearch = true;
+        for (size_t i = Arity; i-- > 0;) {
```

Comment:
Isn't this an equivalent for loop? If so, the previous form is clearer.

Reply:
The previous for loop has a potential overflow risk, causing an infinite loop. It was there intended as I wanted to see if the program goes into the inner condition statement.

The current one doesn't have overflow issue. Combined with an assertion test is better to explore any potential issue.